### PR TITLE
feat: add visibility select in ShareMemoDialog

### PR DESCRIPTION
**Pull request by moehanabi**
I am not familiar with ts/react coding, so please make a double check before merge.

In ShareMemoDialog, user can change the visibility of the memo, so that the memo can be set to public to be viewed by anyone with the link.

related issue: #2868
related pr: #2931

### CODE REVIEW

File: web/src/components/ShareMemoDialog.tsx [hunk 1]
```diff
@@ -75,6 +82,22 @@ const ShareMemoDialog: React.FC<Props> = (props: Props) => {
...
+          id: memo.id,
+          visibility: visibility,
+        },
+        ["visibility"],
+      )
+      .then((newMemo: Memo) => {
+        if (newMemo.visibility == visibility) {
+          toast.success(t("common.changed"));
+        }
+      });
```
Comment (boojack) [inline web/src/components/ShareMemoDialog.tsx:98]: Don't need to use `then` in `async/await` function.

```suggestion
    const updatedMemo = await memoStore
      .updateMemo(
        {
          id: memo.id,
          visibility: visibility,
        },
        ["visibility"],
      )
      .then((newMemo: Memo) => {
        if (newMemo.visibility == visibility) {
          toast.success(t("common.changed"));
        }
      });
```

**Review by boojack — APPROVED** [review]
LGTM

File: web/src/components/ShareMemoDialog.tsx (hunk 1 above)
Comment (moehanabi) [reply web/src/components/ShareMemoDialog.tsx]: I force-pushed a new version, please check it.

**Review by boojack — APPROVED** [review]
LGTM

**Pull request merged**